container/shim/src/modules/registration.js

```diff
@@ -139,6 +140,30 @@ export async function register(initial = false) {
       }
     }
 
+    const certString = certBuffer.toString();
+    const cert = certString.substring(0, certString.indexOf("-----END CERTIFICATE-----") + 25);
+    const caCert = certString.substring(certString.indexOf("-----END CERTIFICATE-----") + 25);
```

Comment:
Use variables to stay DRY and avoid magic number `25`

```suggestion
const boundary = "-----END CERTIFICATE-----"
const boundaryEndIndex = certString.indexOf(boundary) + boundary.length
const cert = certString.substring(0, boundaryEndIndex);
const caCert = certString.substring(boundaryEndIndex);
```

Reply:
Is it really worth it? It's so specific to the string itself, not an external value, it's the length of ----END CERT...

Reply:
My thinking:

`certString.indexOf("-----END CERTIFICATE-----") + 25` is just pasted twice.. that should at least be put into a variable. It's an easy win for code clarity, there's no downside.

`const boundaryEndIndex = certString.indexOf("-----END CERTIFICATE-----") + 25`

Now, what is `25`? My first guess is its the length of `-----END CERTIFICATE-----`, but I had to spend a few seconds confirming it. Save everyone a few seconds by putting it into a descriptive variable name, so I end up with 

```js
const boundary = "-----END CERTIFICATE-----"
const boundaryEndIndex = certString.indexOf(boundary) + boundary.length
```

It might not be worth it for just these 2 lines alone, but this sort of "is it really worth it" mentality can spread over the entire code base and build up technical debt over time. It's better to just do it clean the first time. Make it readable for someone who's just skimming the code.